### ISSUE TITLE
"/.well-known/"  (URI)

### ISSUE THREAD

**Issue by ionutvilie** [kataras/iris#803]
Hi, 

i am playing with the framework and it seems that when using URI paths that starts with `.` it returns 404. (http://www.rfc-editor.org/rfc/rfc5785.txt)

example:
 - ["/" , "/ping"]  --> works
 - "/.well-known" --> 404 

```go
func newApp() *iris.Application {
	app := iris.New()
	est := new(EndpointStore)

	app.Use(recover.New())
	app.Use(logger.New())

	// Method:   GET
	// Resource: http://localhost:8080
	// app.StaticEmbedded("/{dummy:string}", "./static", Asset, AssetNames)
	// app.StaticWeb("/web", "./build")
	// app.Handle("GET", "/", func(ctx iris.Context) {
	// 	ctx.HTML("<h1>Welcome</h1>")
	// 	ctx.
	// })

	app.OnErrorCode(404, func(ctx iris.Context) {
		ctx.Exec("GET", "/")
	})

	app.RegisterView(iris.HTML("./build", ".html").Binary(Asset, AssetNames))

	app.Get("/", func(ctx iris.Context) {
		ctx.ViewData("Page", page)
		ctx.View("index.html")
	})

	assetHandler := app.StaticEmbeddedHandler("./build", Asset, AssetNames)
	// as an alternative of SPA you can take a look at the /routing/dynamic-path/root-wildcard
	// example too
	app.SPA(assetHandler)

	app.Get("/ping", func(ctx iris.Context) {
		ctx.WriteString("pong")
	})
	app.Get("/.well-known", func(ctx iris.Context) {
		ctx.WriteString("OK")
	})
	app.Get(".well-known/ready", func(ctx iris.Context) {
		ctx.WriteString("OK")
	})
	app.Get(".well-known/live", func(ctx iris.Context) {
		ctx.WriteString("OK")
	})
	app.Get(".well-known/metrics", func(ctx iris.Context) {
		ctx.StatusCode(200)
	})
	return app
}

func main() {
	app := newApp()

	app.Run(iris.Addr(":8080"))
}

```

**Comment by kataras** [comment]
hey @ionutvilie I can't reproduce your theory, `.` doesn't gives `404` by itself, router works fine I know every signle word of the http RFC, including commas  :)

```go
package main

import "github.com/kataras/iris"

func newApp() *iris.Application {
	app := iris.New()

	app.OnErrorCode(404, func(ctx iris.Context) {
		ctx.Writef("404")
	})

	app.Get("/", func(ctx iris.Context) {
		ctx.Writef("index")
	})

	app.Get("/ping", func(ctx iris.Context) {
		ctx.WriteString("pong")
	})
	app.Get("/.well-known", func(ctx iris.Context) {
		ctx.WriteString("well-known")
	})
	app.Get(".well-known/ready", func(ctx iris.Context) {
		ctx.WriteString("read")
	})
	app.Get(".well-known/live", func(ctx iris.Context) {
		ctx.WriteString("live")
	})
	app.Get(".well-known/metrics", func(ctx iris.Context) {
		ctx.Writef("metrics")
	})
	return app
}

func main() {
	app := newApp()

	app.Run(iris.Addr(":8080"))
}
```

![nobug1](https://user-images.githubusercontent.com/22900943/32553817-939153b6-c4a0-11e7-8057-22e4060956a6.png)
![nobug2](https://user-images.githubusercontent.com/22900943/32553818-93b46504-c4a0-11e7-9bad-f51d1cc1a15c.png)

Try to give us a more complete example in order to find what's wrong with your app, thank you very much for your time to open an issue, I will wait!

**Comment by ionutvilie** [comment]
:) you are right, if i disable the SPA handler the path works fine.

```go
	assetHandler := app.StaticEmbeddedHandler("./build", Asset, AssetNames)
	// as an alternative of SPA you can take a look at the /routing/dynamic-path/root-wildcard
	// example too
	app.SPA(assetHandler)
```

[bindata.go](https://1drv.ms/u/s!Au5GeigUluF3gexY9LoRQhZ5ye_K4w)

btw i am not sure if i have correctly used the handlers, 
it is more of a copy, use what i can for a proof of concept. :)

**Comment by kataras** [comment]
`SPA` is a built'n router wrapper, it runs even before the router and it is calling the router when no asset/static file was found; https://github.com/kataras/iris/blob/master/core/router/router_spa_wrapper.go#L71 , otherwise it serves the file; https://github.com/kataras/iris/blob/master/core/router/router_spa_wrapper.go#L90. But `SPA` has also a default validator on "what is an asset" and that is if contains a dot `.`; https://github.com/kataras/iris/blob/master/core/router/router_spa_wrapper.go#L39 , the problem is exactly there, so you can fix the issue by initializating the [SPABuilder](https://github.com/kataras/iris/blob/master/core/router/router_spa_wrapper.go#L16) manually and change its `AssetValidators` or we can find a better way to handle this, give me time on this it's just a API-perspective thing

**Comment by kataras** [comment]
@ionutvilie Fixed on the new release `v8.5.8` with https://github.com/kataras/iris/commit/d77c17336750a4aea43dc55760d4509dff88cbdc#diff-9f822a50b8f642f554819681532b7b7e . Update and you should be fine, thanks for the report, it was really an issue.

**Comment by ionutvilie** [comment]
cheers, thank you very much for the quick fix. :)
